src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs

```diff
@@ -365,22 +365,12 @@ public TransactionNode(Transaction tx)
         /// <param name="connectToNode">The node that will be connected to.</param>
         public static void Connect(CoreNode thisNode, CoreNode connectToNode)
         {
-            var cancellation = new CancellationTokenSource();
-            cancellation.CancelAfter(TimeSpan.FromSeconds(30));
+            if (IsNodeConnectedTo(thisNode, connectToNode))
+                return;
 
-            WaitLoop(() =>
```

Comment:
OK so the idea here was to retry a few times if the initial connection could not be made, but maybe it was a bit overkill.